common/src/main/java/com/wynntils/models/npcdialogue/event/NpcDialogueProcessingEvent.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © Wynntils 2022-2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.npcdialogue.event;
+
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.npcdialogue.type.NpcDialogue;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import net.minecraft.network.chat.Component;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * Event that is fired at different stages during while processing an NPC dialogue.
+ */
+public abstract class NpcDialogueProcessingEvent extends Event {
+    /**
+     * Event that is fired before processing an NPC dialogue.
+     * This event can be used to add any pre-processing to the dialogue.
+     */
+    public static class Pre extends NpcDialogueProcessingEvent {
+        private final NpcDialogue dialogue;
+
+        // This is a future that can be used to add any post-processing to the dialogue.
+        // This is used for things like translation, transcription, etc.
+        private CompletableFuture<List<StyledText>> postProcessedDialogueFuture;
```

Comment:
I assume this is since translation takes time. Nevertheless, we have not really used async code in Artemis, and I'm a bit scared about what that means. I guess this is just exposing something that already existed in the translation service, but nevertheless, it looks like this is creating more exposure for it.

Reply:
I understand. We'll just have to take care when working with this.